core/validatorapi/validatorapi.go

```diff
@@ -249,6 +249,9 @@ func (c Component) SubmitAttestations(ctx context.Context, attestations []*eth2p
 
 		pubkey, err := c.pubKeyByAttFunc(ctx, slot, int64(att.Data.Index), int64(indices[0]))
 		if err != nil {
+			log.Error(ctx, "Failed to find pubkey for the given attestation", err, z.I64("slot", slot),
```

Comment:
never log errors and return errors. Rather wrap the error and add the structured fields